iree/compiler/Dialect/VMLA/IR/VMLAOps.td

```diff
@@ -178,6 +178,12 @@ def VMLA_TransposeOp : VMLA_ElementTypeOp<"transpose", [VMLA_IncludeShapes]> {
     VMLA_Shape:$dst_shape,
     VMLA_AnyTypeAttr:$element_type
   );
+
+  let assemblyFormat = [{
+    $src`,` $src_shape`,` $dst`,` $dst_shape attr-dict `:`
+    `(`type($src_shape)`,` type($dst_shape)`)`
+    $element_type
```

Comment:
I think this would actually require nontrivial work because while we know the element type, we don't know the shape without using info from the other operands and I think the declarative parser type inference isn't advanced enough for this (I can't even get it to let me elide the attr type when it's the same as an operand type). I complained to River about some of these limitations though, so maybe it will get better soon :-) I think if we send him more examples, he'll implement more goodness :-D